cps/src/main/java/org/jenkinsci/plugins/workflow/cps/steps/ParallelStep.java

```diff
@@ -45,17 +51,29 @@ public StepExecution start(StepContext context) throws Exception {
         return new ParallelStepExecution(this, context);
     }
 
+    /*package*/ boolean isFailFast() {
+        return failFast;
+    }
+
+
     @PersistIn(PROGRAM)
     static class ResultHandler implements Serializable {
         private final StepContext context;
+        private final ParallelStepExecution stepExecution;
```

Comment:
`StepExecution` is persisted in program state, so this is OK. `BodyExecution`s are stored to allow cancellation of parallel branches.